client/src/components/MoviesContainer.jsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import MoviesResults from './MoviesResults'
+import movies from './movies'
+import SearchContainer from './SearchContainer';
+import searchFilterType from './SearchFilter/searchFilterType';
+import sortType from './Sort/sortType'
+import BackToSearch from './BackToSearch'
+import MoviePage from './MoviePage'
+
+class MoviesContainer extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            list: [],
+            filter: searchFilterType.title,
+            sortType: sortType.releaseDate,
+            selectedMovie:null,
+            moviesOfTheSameGenre:[],
+            hasError:false
+        };
+    }
+
+    componentDidMount() {
+        this.setState({
+            list: this.sort(movies.data, this.state.sortType)
+        });
+    }
+
+    componentDidCatch(error, info){
+        this.setState({hasError:true});
+        console.log("I crashed: " + error +" "+info);
+    }
+
+    onSwitchFilter(type) {
+        if (this.state.filter === type)
+            return;
+
+        this.setState({
+                filter: type
+            });
+    }
+
+    sortByReleaseDate(first, second) {
+        if (new Date(first.release_date) > new Date(second.release_date)) {
+            return -1;
+        }
+
+        if (new Date(first.release_date) < new Date(second.release_date)) {
+            return 1;
+        }
+
+        return 0;
+    }
+
+    sortByRating(first, second) {
+        return second.vote_average - first.vote_average;
+    }
+
+    sort(inMovies, sortBy) {
+        switch (sortBy) {
+            case sortType.rating:
+                return inMovies.sort(this.sortByRating);
```

Comment:
_minor_
`Array.prototype.sort` eventually will mutate array, so it's better to use cloned array object rather than mutating the one that you're passing through.